lib/reqIdGenFactory.js

```diff
@@ -1,16 +1,21 @@
 'use strict'
 
+const numberLookupA = new Array(1296).fill(0).map((v, i) => i.toString(36))
+const numberLookupB = numberLookupA.map(v => v.padStart(2, '0'))
+
 module.exports = function (requestIdHeader, optGenReqId) {
-  // 2,147,483,647 (2^31 − 1) stands for max SMI value (an internal optimization of V8).
-  // With this upper bound, if you'll be generating 1k ids/sec, you're going to hit it in ~25 days.
-  // This is very likely to happen in real-world applications, hence the limit is enforced.
-  // Growing beyond this value will make the id generation slower and cause a deopt.
-  // In the worst cases, it will become a float, losing accuracy.
-  const maxInt = 2147483647
-  let nextReqId = 0
+  let numberLookup = numberLookupA
+  let i = 1
+  let sequence = 'req-'
+  let sequenceNum = 0
+
   function defaultGenReqId (req) {
-    nextReqId = (nextReqId + 1) & maxInt
-    return `req-${nextReqId.toString(36)}`
+    if (i === 1296) {
+      i = 0
+      sequenceNum === 0 && (numberLookup = numberLookupB)
+      sequence = `req-${(++sequenceNum).toString(36)}`
```

Comment:
Out of curiosity: Isn't concatenation slightly faster?

Reply:
Uh. Idk. Do you have a suggestion?

Reply:
nvm, I made benchmarks and template literals were faster 👍